addons/html_editor/static/tests/paste.test.js

```diff
@@ -766,12 +766,270 @@ describe("Simple html elements containing <br>", () => {
+        });
+    });
+    test("should paste all nodes as blockquote when pasting in blockquote", async () => {
+        await testEditor({
+            contentBefore: "<blockquote>[]<br></blockquote>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter:
+                "<blockquote>abc</blockquote><blockquote>def</blockquote><blockquote>ghi[]</blockquote>",
+        });
+        await testEditor({
+            contentBefore: "<blockquote>x[]</blockquote>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter:
+                "<blockquote>xabc</blockquote><blockquote>def</blockquote><blockquote>ghi[]</blockquote>",
+        });
+        await testEditor({
+            contentBefore: "<blockquote>[]x</blockquote>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter:
+                "<blockquote>abc</blockquote><blockquote>def</blockquote><blockquote>ghi[]x</blockquote>",
+        });
+        await testEditor({
+            contentBefore: "<blockquote>x[]y</blockquote>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter:
+                "<blockquote>xabc</blockquote><blockquote>def</blockquote><blockquote>ghi[]y</blockquote>",
+        });
+    });
+    test("should paste all nodes as pre when pasting in pre", async () => {
+        await testEditor({
+            contentBefore: "<pre>[]<br></pre>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter: "<pre>abc</pre><pre>def</pre><pre>ghi[]</pre>",
+        });
+        await testEditor({
+            contentBefore: "<pre>x[]</pre>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter: "<pre>xabc</pre><pre>def</pre><pre>ghi[]</pre>",
+        });
+        await testEditor({
+            contentBefore: "<pre>[]x</pre>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter: "<pre>abc</pre><pre>def</pre><pre>ghi[]x</pre>",
+        });
+        await testEditor({
+            contentBefore: "<pre>x[]y</pre>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter: "<pre>xabc</pre><pre>def</pre><pre>ghi[]y</pre>",
+        });
+    });
+});
```

Comment:
Since `pre` is meant to write code (we even call the format "code" in the UI if I remember correctly, I would expect to actually see the html code that I'm pasting. Otherwise, how can you copy paste some code from your IDE to a task description? You can't and you have to rewrite all your code by hand? @lebl-odoo What do you think?

Reply:
I agree with @dmo-odoo that they (blockquote & pre) should be considered like wrappers. We should keep styles for those case & apply the other rules when relevant.

Reply:
@dmo-odoo How about I drop this part of the code from the PR? And create another PR fixing this issue as this is a forward port PR which originated from 16.0.

Reply:
@deso-odoo If it's easier for you, sure, as long as the behavior in the end is the right one, it's good for me 😉